### DISCUSSION TITLE
404 Page

### DISCUSSION THREAD

**Discussion by siddhida** [vercel/next.js#22484] [Help]
In my projects there are more than 2000 routes. This included mix of server side rendering and client side rendering. I have made one 404 page. Which works fine with routing pattern but I need to render 404 page even when user enters wrong. 
for e.g , right route : siddhi.com/roses
and e.g for  wrong route: siddhi.com/rose

How can we achieve that?

**Comment by indegser** [comment]
I'm not sure how your pages folder is structured.
In my case, I usually create HOC component for checking authorized(allowed) urls.

```tsx
// Some hoc component e.g <AllowOrNotFound />
// Validate allowed pages with blacklist
const router = useRouter();
const blacklist = ['/roses', ...];
const isNotAllowed = blacklist.includes(router.asPath);
if (isNotAllowed) return <NotFound /> // Your 404 component
return <Component />
```

```tsx
// pages/*.tsx
return (
  <AllowOrNotFound>
    <Roses /> // Your component
  </AllowOrNotFound>
)
```
